enterprise/signals.py

```diff
@@ -106,6 +107,59 @@ def default_content_filter(sender, instance, **kwargs):     # pylint: disable=un
         instance.save()
 
 
+@receiver(post_save, sender=EnterpriseCustomerUser)
+def assign_or_delete_enterprise_admin_role(sender, instance, **kwargs):     # pylint: disable=unused-argument
+    """
+    Assign or delete enterprise_admin role for EnterpriseCustomerUser when updated.
+
+    This only occurs if a PendingEnterpriseCustomerAdminUser record exists.
+    """
+    if instance.user:
+        enterprise_admin_role, __ = SystemWideEnterpriseRole.objects.get_or_create(name=ENTERPRISE_ADMIN_ROLE)
+        try:
+            pending_enterprise_admin_user = PendingEnterpriseCustomerAdminUser.objects.get(
+                user_email=instance.user.email,
+                enterprise_customer=instance.enterprise_customer,
+            )
+        except PendingEnterpriseCustomerAdminUser.DoesNotExist:
+            pending_enterprise_admin_user = None
+
+        if kwargs['created'] and pending_enterprise_admin_user:
+            # EnterpriseCustomerUser record was created and a pending admin user
+            # exists, so assign the enterprise_admin role.
+            pending_enterprise_admin_user.activate_admin_permissions(
+                user=instance.user,
+                enterprise_customer=instance.enterprise_customer,
+            )
+        elif not kwargs['created'] and not instance.linked:
```

Comment:
Should we have an else statement with logging for if anything ends up in a weird state that doesn't fit the current two branches?